src/twisted/web/test/test_web.py

```diff
@@ -777,6 +777,25 @@ def test_encoding(self):
                           zlib.decompress(body, 16 + zlib.MAX_WBITS))
 
 
+    def test_whitespaceInAcceptEncoding(self):
+        """
+        If the client request passes a I{Accept-Encoding} header which mentions
+        gzip, with whitespace inbetween the encoding name and the commas,
+        L{server._GzipEncoder} automatically compresses the data.
+        """
+        request = server.Request(self.channel, False)
+        request.gotLength(0)
+        request.requestHeaders.setRawHeaders(b"Accept-Encoding",
+                                             [b"deflate, gzip"])
```

Comment:
I find the current test code hard to read... why we use gzip and not deflate?

I know that we only support .gz and .bz2 ... but the test should not make this assumption,

We might later add support for deflate, and then this test might fail ... and then you might not know why